sphinx_gallery/gen_gallery.py

```diff
@@ -329,6 +351,22 @@ def sumarize_failing_examples(app, exception):
                          "\n" + "-" * 79)
 
 
+def check_duplicate_filenames(files):
+    """Check for duplicate filenames across gallery directories."""
+    # Check whether we'll have duplicates
+    used_names = set()
+    dup_names = list()
+    for this_file in files:
+        this_fname = os.path.basename(this_file)
+        if this_fname in used_names:
+            dup_names.append(this_file)
+        used_names = used_names.union([this_fname])
+
+    if len(dup_names) > 0:
+        warn('Duplicate file name(s) found. Having duplicate file names will '
```

Comment:
how do we check for this with `pytest`? The typical `with pytest.warns` doesn't work. I can't figure out a way to see which logs have been filed using the `logger` object...anybody have an idea?

Reply:
if it's a standard Python logger you can add a handler along these lines:

https://stackoverflow.com/questions/9534245/python-logging-to-stringio-handler

Or probably better, see if something like this would work:

https://pypi.python.org/pypi/pytest-logger

If it's not a standard logger, you could maybe hack it by temporarily (`try`/`except`) changing `sys.stdout` but I've seen this have undesired consequences with testing libraries before